src/app/modules/item/pages/item-edit-advanced-parameters/item-edit-advanced-parameters.component.ts

```diff
@@ -58,7 +60,16 @@ export class ItemEditAdvancedParametersComponent implements OnInit {
     value: 'None'
   }];
 
-  constructor() { }
+  attemptId$ = this.itemDataSource.state$.pipe(
+    map(state => {
+      if (state.isReady) {
+        return state.data.route.attemptId || state.data.route.parentAttemptId;
+      }
+      return undefined;
+    })
+  );
+
+  constructor(private itemDataSource: ItemDataSource) { }
```

Comment:
OK, what I wrote was unclear, sorry about that.
What I meant is: the item datasource service should only be injected into pages (or routes) declared in the `item-routing.module`. Currently there are only 3: `ItemById`, `ItemDetails` and `ItemEdit`

TL;DR: Here it should be `ItemEdit`

That's because currently the datasource services are currently a workaround to share data between `ItemById` and its sub routes `ItemDetails` and `ItemEdit`. At the time, we didn't know there was a way to pass data between a route and its sub routes. Now that we know how to do it, Damien intends to remove those services in a near future.